updated some incorrect info (#3088)
hammerandchisel/discord-api-docs: docs/topics/OAuth2.md

1. Bots are added to guilds through the OAuth2 API, and cannot accept normal invites.
2. Bots cannot have friends, nor be added to or join Group DMs.

> 3. Bots do not have a maximum number of Guilds (unlike user accounts, which are limited to 100 for non-nitro users, and to 200 for nitro users).

Reviewer: ```suggestion
3. Bots do not have a maximum number of Guilds, unless they are unverified. These are unlike user accounts that have a maximum of 100, and Nitro subscribers that have 200.
```
